File: python/infinicore/ops/deepseek_v4_rms_norm.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor


def deepseek_v4_rms_norm(input: Tensor, weight: Tensor, epsilon: float, *, out: Tensor | None = None) -> Tensor:
    if out is None:
        return Tensor(_infinicore.deepseek_v4_rms_norm(input._underlying, weight._underlying, epsilon))
    _infinicore.deepseek_v4_rms_norm_(out._underlying, input._underlying, weight._underlying, epsilon)
    return out

File: src/infinicore/ops/deepseek_v4_rms_norm/deepseek_v4_rms_norm.cc
#include "infinicore/ops/deepseek_v4_rms_norm.hpp"

#include "infinicore/ops/rms_norm.hpp"

namespace infinicore::op {

Tensor deepseek_v4_rms_norm(const Tensor &x, const Tensor &weight, float epsilon) {
    return rms_norm(x, weight, epsilon);
}

void deepseek_v4_rms_norm_(Tensor y, const Tensor &x, const Tensor &weight, float epsilon) {
    rms_norm_(y, x, weight, epsilon);
}

} // namespace infinicore::op
